monai/apps/nuclick/transforms.py

```diff
@@ -0,0 +1,469 @@
+        rem_sm = morphology.remove_small_objects(img_np.astype(bool), min_size=min_size)
+        mask_percentage = self.mask_percent(rem_sm)
+        if (mask_percentage >= overmask_thresh) and (min_size >= 1) and (avoid_overmask is True):
+            new_min_size = round(min_size / 2)
+            rem_sm = self.filter_remove_small_objects(img_np, new_min_size, avoid_overmask, overmask_thresh)
+        return rem_sm
+
+
+class AddPointGuidanceSignald(RandomizableTransform):
+    """
+    Adds Guidance Signal to the input image
+
+    Args:
+        image: source image
+        label: source label
+        others: source others (other labels from the binary mask which are not being used for training)
+        drop_rate:
+        jitter_range: noise added to the points in the point mask for exclusion mask
+    """
+
+    def __init__(
+        self,
+        image: str = "image",
+        label: str = "label",
+        others: str = "others",
+        drop_rate: float = 0.5,
+        jitter_range: int = 3,
+    ):
+        super().__init__()
+
+        self.image = image
+        self.label = label
+        self.others = others
+        self.drop_rate = drop_rate
+        self.jitter_range = jitter_range
+
+    def __call__(self, data):
+        d = dict(data)
+
+        image = d[self.image]
+        mask = d[self.label]
+        others = d[self.others]
+
+        inc_sig = self.inclusion_map(mask[0])
+        exc_sig = self.exclusion_map(others[0], drop_rate=self.drop_rate, jitter_range=self.jitter_range)
+
+        image = np.concatenate((image, inc_sig[np.newaxis, ...], exc_sig[np.newaxis, ...]), axis=0)
+        d[self.image] = image
+        return d
+
+    def inclusion_map(self, mask):
+        point_mask = np.zeros_like(mask)
+        indices = np.argwhere(mask > 0)
+        if len(indices) > 0:
+            idx = np.random.randint(0, len(indices))
+            point_mask[indices[idx, 0], indices[idx, 1]] = 1
+
+        return point_mask
+
+    def exclusion_map(self, others, jitter_range=3, drop_rate=0.5):
+        point_mask = np.zeros_like(others)
+        if drop_rate == 1.0:
+            return point_mask
+
+        max_x = point_mask.shape[0] - 1
+        max_y = point_mask.shape[1] - 1
+        stats = measure.regionprops(others)
+        for stat in stats:
+            x, y = stat.centroid
+            if np.random.choice([True, False], p=[drop_rate, 1 - drop_rate]):
+                continue
+
+            # random jitter
+            x = int(math.floor(x)) + random.randint(a=-jitter_range, b=jitter_range)
+            y = int(math.floor(y)) + random.randint(a=-jitter_range, b=jitter_range)
+            x = min(max(0, x), max_x)
+            y = min(max(0, y), max_y)
+            point_mask[x, y] = 1
+
+        return point_mask
+
+
+class AddClickSignalsd(Transform):
+    def __init__(self, image, foreground="foreground", bb_size=128):
```

Comment:
Please add doc-string for the args and also describe the expected "location" field in the data.

Thanks.

Reply:
Added